core/src/main/java/org/apache/accumulo/core/file/streams/PositionedOutputs.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.core.file.streams;
+
+import java.io.FilterOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import org.apache.hadoop.fs.FSDataOutputStream;
+
+public class PositionedOutputs {
+  public static PositionedOutputStream wrap(final OutputStream fout) {
+    if (fout instanceof FSDataOutputStream) {
+      return new PositionedOutputStream(fout) {
+        @Override
+        public long position() throws IOException {
+          return ((FSDataOutputStream) fout).getPos();
+        }
+      };
+    } else if (fout instanceof PositionedOutput) {
+      return new PositionedOutputStream(fout) {
+        @Override
+        public long position() throws IOException {
+          return ((PositionedOutput) fout).position();
+        }
+      };
+    } else {
+      return new PositionedOutputStream(fout) {
+        @Override
+        public long position() throws IOException {
+          throw new UnsupportedOperationException("Underlying stream does not support position()");
+        }
+      };
+    }
+  }
+
+  public static abstract class PositionedOutputStream extends FilterOutputStream implements PositionedOutput {
```

Comment:
What I really want to say is "This method returns an `OutputStream` which implements `PositionedOutput`"  But there's not really a way to say that in Java.

I'm probably just being too clever with `PositionedOutput` anyways.  The underlying issue is that `BCFile.Writer` needs an `OutputStream` implementing `DataOutput` with some facsimile of the `getPos()` method from `FSDataOutputStream`.  `RateLimitedOutputStream` doesn't want to care about or interfere with either of these.  But there doesn't seem to be any way to separate concerns cleanly.  `getPos()` doesn't have an interface (like `Seekable`), and `java.io.DataOutputStream` (and so `FSDataOutputStream`) isn't implemented in a way which readily supports subclassing/inheritance.